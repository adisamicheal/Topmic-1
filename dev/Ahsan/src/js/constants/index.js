export const FILL_ME = 'FILL_ME';