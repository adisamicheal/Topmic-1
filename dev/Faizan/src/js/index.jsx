import '../css/styles.scss'
import ReactDOM from 'react-dom';
import React from 'react';
import Contact from './Contact';

ReactDOM.render(
  <Contact />,
  document.querySelector('#app')
);