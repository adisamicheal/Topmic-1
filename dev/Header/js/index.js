// Below for css files which will be bundled to styles.css
import '../css/styles.scss'
import '../../node_modules/bootstrap/dist/css/bootstrap.css'
import '../../node_modules/bootstrap/dist/css/bootstrap-theme.css'

import ReactDOM from 'react-dom';
import React from 'react';
import AppNavSection  from './app';

// Below happen under the hood when initializing a custom Element
// var obj = new App()
// obj.props = {}
// obj.render()

//Now when setState is used to change the data then below happens
// state has changed using setState
// obj.render() - this will re-render the virtual dom and custom elements
                  //resulting in props being re-sent


ReactDOM.render(
  <AppNavSection/>,
  document.querySelector('#winsome')
  );