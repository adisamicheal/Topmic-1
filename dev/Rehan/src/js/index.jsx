import ReactDOM from 'react-dom';
import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import Achievers from './components/Achievers';
ReactDOM.render(
		<Achievers />,
  document.querySelector('#app')
);