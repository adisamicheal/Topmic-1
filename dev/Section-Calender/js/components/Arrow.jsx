import React from 'react';

const Arrow=({onClickLeftArrow,onClickRightArrow})=>{
	return (
		<div>
			<div className="arrowRight"
			onClick={onClickRightArrow}
			>
			
			</div>
			<div className="arrowLeft"
			onClick={onClickLeftArrow}
			>
			</div>
		</div>
	)
}

export default Arrow;