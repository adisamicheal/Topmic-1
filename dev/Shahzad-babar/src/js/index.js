import '../css/styles.scss'
import React from 'react';
import ReactDOM from 'react-dom';
import Juniors from './Juniors';

ReactDOM.render(
  <Juniors />,
  document.querySelector('#app'),
  
);