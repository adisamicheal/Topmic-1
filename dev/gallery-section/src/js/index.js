import '../css/styles.scss'
import '../../node_modules/bootstrap/dist/css/bootstrap.css'
console.log(`Loading...`);
import ReactDOM from 'react-dom';
import React from 'react';
import App from './app';

ReactDOM.render(
	<App />,
	document.querySelector('#app'),
	function(){console.log('React App...')}
);