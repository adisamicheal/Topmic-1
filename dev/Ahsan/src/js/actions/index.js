import { FILL_ME } from '../constants';
