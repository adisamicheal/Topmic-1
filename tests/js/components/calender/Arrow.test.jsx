import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Arrow } from '../../../../src/js/components/calender/Arrow';
import * as constants from '../../../../src/js/constants';
import * as actions from '../../../../src/js/actions';

Enzyme.configure({ adapter: new Adapter() });

describe('Test the Arrow Component', () => {
	const onClickLeftArrow = jest.fn();
	const onClickRightArrow = jest.fn();

	const wrapper = shallow(<Arrow />);

	it('checks the rendring of the Arrow component', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('checks the Left Click', () => {
		wrapper.find('.arrowLeft').simulate('click');
		expect(onClickLeftArrow.mock.calls.length).toBe(0);
	});
	it('checks the Right Click', () => {
		wrapper.find('.arrowRight').simulate('click');
		expect(onClickRightArrow.mock.calls.length).toBe(0);
	});
	it('checks the Left Enter key Press', () => {
		wrapper.find('.arrowLeft').simulate('onkeypress');
		expect(onClickLeftArrow.mock.calls.length).toBe(0);
	});
	it('checks the Right Click', () => {
		wrapper.find('.arrowRight').simulate('onkeypress');
		expect(onClickRightArrow.mock.calls.length).toBe(0);
	});
	it('should test the onClickRightArrow function', () => {
		const expectedActionRightArrow = {
			type: constants.ARROW_RIGHT,
			payload: 1
		};
		expect(actions.onClickRightArrow()).toEqual(expectedActionRightArrow);
	});
	it('should test the onClickLeftArrow function', () => {
		const expectedActionLeftArrow = {
			type: constants.ARROW_LEFT,
			payload: 1
		};
		expect(actions.onClickLeftArrow()).toEqual(expectedActionLeftArrow);
	});
	it('test the if condition',()=>{
		const e={
			nativeEvent:{
				keyCode:13
			}
		}
		expect(e.nativeEvent.keyCode).toBe(13)
	})
});
