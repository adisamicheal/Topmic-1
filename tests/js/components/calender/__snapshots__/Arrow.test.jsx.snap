// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Arrow Component checks the rendring of the Arrow component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Arrow
    onClickLeftArrow={null}
    onClickRightArrow={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="arrowRight"
          onClick={null}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        />,
        <div
          className="arrowLeft"
          onClick={null}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "arrowRight",
          "onClick": null,
          "onKeyPress": [Function],
          "role": "button",
          "tabIndex": "0",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "arrowLeft",
          "onClick": null,
          "onKeyPress": [Function],
          "role": "button",
          "tabIndex": "0",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="arrowRight"
            onClick={null}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          />,
          <div
            className="arrowLeft"
            onClick={null}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "arrowRight",
            "onClick": null,
            "onKeyPress": [Function],
            "role": "button",
            "tabIndex": "0",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "arrowLeft",
            "onClick": null,
            "onKeyPress": [Function],
            "role": "button",
            "tabIndex": "0",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
